### PR TITLE
Introduce store type v2 flag

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by dzajkowski** [comment]
bors try

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala [hunk 1]
```diff
@@ -449,6 +450,22 @@ object Runtime {
                       ](context, Branch.REPLAY)
       } yield ((context, space, replaySpace))
 
+    def createSpace2(
```
Comment (lukasz-golebiewski) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: `createNextSpace`?

File: rspace/src/main/scala/coop/rchain/rspace/Context.scala [hunk 2]
```diff
@@ -81,6 +81,18 @@ private[rspace] class MixedContext[F[_], C, P, A, K] private[rspace] (
   }
 }
 
+class NoopContext[F[_], C, P, A, K] extends Context[F, C, P, A, K] {
```
Comment (ArturGajowy) [inline rspace/src/main/scala/coop/rchain/rspace/Context.scala]: Looks like a lost opportunity for `NopeContext` :stuck_out_tongue:

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala [hunk 3]
```diff
@@ -527,9 +554,16 @@ object RSpace {
       _ <- checkCreateDir(rootsStore.path)
       historyReader <- HistoryRepositoryInstances
                         .lmdbRepository[F, C, P, A, K](config)
-      cache <- Cell.refCell[F, Cache[C, P, A, K]](Cache())
-      store = HotStore.inMem(Sync[F], cache, historyReader, cp, ck)
+      store <- inMemoryStore(historyReader)
```
Comment (ArturGajowy) [inline rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala]: :ok_hand:

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala [hunk 4]
```diff
@@ -468,6 +469,32 @@ object RSpace {
 
   }
 
+  def createWithReplay[F[_], C, P, A, R, K](dataDir: Path, mapSize: Long)(
```
Comment (ArturGajowy) [inline rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala]: this looks unused, but I assume the usage will come later? Maybe it'd make sense to commit that with the usage? Or a test at least?

**Comment by ArturGajowy** [comment]
bors try

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala (hunk 4 above)
Comment (dzajkowski) [reply rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala]:
```
F[(RhoContext[F], RhoISpace[F], RhoReplayISpace[F])] =
      for {
        withReplay <- NextRSpace.createWithReplay[
```?

Comment (ArturGajowy) [reply rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala]: lol :D

**Pull request merged**